lib/rules/id-length.js

```diff
@@ -40,6 +42,10 @@ module.exports = function(context) {
             return parent.id === node;
         },
         "Property": function(parent, node) {
+            if (!properties) {
```

Comment:
Why not

```js
    return properties !== false && parent.key === node;
```

instead?

Alternatively you can override this function and set `SUPPORTED_EXPRESSIONS.Property` to `false` if `properties` is `false`.